cpp/src/io/orc/reader_impl.hpp

```diff
@@ -91,143 +63,27 @@ class reader::impl {
-   * @param row_groups Vector of list of row index descriptors
-   * @param out_buffers Column buffers for columns.
-   * @param list_col Vector of column metadata of list type parent columns.
-   * @param level Current nesting level being processed.
-   */
-  void aggregate_child_meta(cudf::detail::host_2dspan<gpu::ColumnDesc> chunks,
-                            cudf::detail::host_2dspan<gpu::RowGroup> row_groups,
-                            std::vector<column_buffer>& out_buffers,
-                            std::vector<orc_column_meta> const& list_col,
-                            int32_t const level);
-
-  /**
-   * @brief Assemble the buffer with child columns.
-   *
-   * @param orc_col_id Column id in orc.
-   * @param col_buffers Column buffers for columns and children.
-   * @param level Current nesting level.
-   */
-  column_buffer&& assemble_buffer(size_type const orc_col_id,
-                                  std::vector<std::vector<column_buffer>>& col_buffers,
-                                  size_t const level,
-                                  rmm::cuda_stream_view stream);
-
-  /**
-   * @brief Create columns and respective schema information from the buffer.
-   *
-   * @param col_buffers Column buffers for columns and children.
-   * @param out_columns Vector of columns formed from column buffers.
-   * @param schema_info Vector of schema information formed from column buffers.
-   * @param stream CUDA stream used for device memory operations and kernel launches
-   */
-  void create_columns(std::vector<std::vector<column_buffer>>&& col_buffers,
-                      std::vector<std::unique_ptr<column>>& out_columns,
-                      std::vector<column_name_info>& schema_info,
-                      rmm::cuda_stream_view stream);
-
-  /**
-   * @brief Create empty columns and respective schema information from the buffer.
-   *
-   * @param col_buffers Column buffers for columns and children.
-   * @param schema_info Vector of schema information formed from column buffers.
-   * @param stream CUDA stream used for device memory operations and kernel launches
-   *
-   * @return An empty column equivalent to orc column type.
-   */
-  std::unique_ptr<column> create_empty_column(size_type const orc_col_id,
-                                              column_name_info& schema_info,
-                                              rmm::cuda_stream_view stream);
-
-  /**
-   * @brief Setup table for converting timestamp columns from local to UTC time
-   *
-   * @return Timezone table with timestamp offsets
-   */
-  std::unique_ptr<table> compute_timezone_table(
-    std::vector<cudf::io::orc::metadata::stripe_source_mapping> const& selected_stripes,
-    rmm::cuda_stream_view stream);
-
- private:
-  rmm::mr::device_memory_resource* _mr = nullptr;
-  std::vector<std::unique_ptr<datasource>> _sources;
+  std::vector<std::unique_ptr<datasource>> const _sources;  // Unused but owns data for others
```

Comment:
This comment does not match

Reply:
Actually this is correct. `_sources` are not used anywhere but members of `_metadata` has pointers refer to it and will be accessed later.

Reply:
I'll update the comment.

Reply:
That's surprising, did not realize we always use sources through metadata objects :)